internal/integration/spec.go

```diff
@@ -0,0 +1,114 @@
+// Package integration ...
+// Copyright 2019 New Relic Corporation. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+package integration
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path"
+	"regexp"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	yaml "gopkg.in/yaml.v2"
+)
+
+const fileNameMatcher = `^prometheus_.*\.yml$`
+
+// Specs contains all the services specs mapped with the service name
+type Specs struct {
+	SpecsByName map[string]Spec
+}
+
+// Spec contains the rules to group metrics into entities
+type Spec struct {
+	Provider string      `yaml:"provider"`
+	Service  string      `yaml:"service"`
+	Entities []EntityDef `yaml:"entities"`
+}
+
+// EntityDef has info related to each entity
+type EntityDef struct {
+	Type       string        `yaml:"name"`
+	Properties PropertiesDef `yaml:"properties"`
+}
+
+// PropertiesDef defines the dimension used to get entity names
+type PropertiesDef struct {
+	Dimensions []string `yaml:"dimensions"`
+}
+
+// LoadSpecFiles loads all service spec files named like "prometheus_*.yml" that are in the filesPath
+func LoadSpecFiles(filesPath string) (Specs, error) {
+	specs := Specs{SpecsByName: make(map[string]Spec)}
+	var files []string
+
+	filesInPath, err := ioutil.ReadDir(filesPath)
+	if err != nil {
+		return specs, err
+	}
+	for _, f := range filesInPath {
+		if ok, _ := regexp.MatchString(fileNameMatcher, f.Name()); ok {
+			files = append(files, path.Join(filesPath, f.Name()))
+		}
+	}
+
+	for _, file := range files {
+		f, err := ioutil.ReadFile(file)
+		if err != nil {
+			logrus.Errorf("fail to read service spec file %s: %s ", file, err)
+			continue
+		}
+
+		var sd Spec
+		err = yaml.Unmarshal(f, &sd)
+		if err != nil {
+			logrus.Errorf("fail parse service spec file %s: %s", file, err)
+			continue
+		}
+		logrus.Debugf("spec file loaded for service:%s", sd.Service)
+		specs.SpecsByName[sd.Service] = sd
+	}
+
+	return specs, nil
+}
+
+// getEntity returns entity name and type of the metric based on the spec configuration defined for the service.
+// metric example: serviceName_entityName_metricName{dimension="dim"} 0
```

Comment:
This won't work very well in some cases. For example, in redis we have a metric named **redis_last_slow_execution_duration_seconds** which won't translate very well. The "entity name" here would be "last_slow_execution" but this code will name it "last".
I think we probably need to declare a format or something that can work with more metrics

Reply:
yes this will not work for all the cases, i talked with @morecar about this and we decided to go for this case and put the rest of the metrics in a main entity for each service, so right now all metrics that do not match with an entity described like this will go to the main entity.